### PR TITLE
Feature xrpf permissions

### PR DESCRIPTION
**Pull request by AtifMahmud**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Height Task ID (If applicable):

## Type of change

  - [] Bug fix (non-breaking change which fixes an issue)
  - [x] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - [x] This change requires a documentation update

# Checklist:

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings
  - [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by AtifMahmud** [comment]
Currently Hardware data permissions. Will do other PRs for other data types

File: Runtime/Scripts/FixationRecorder.cs [hunk 1]
```diff
@@ -981,10 +981,13 @@ void RecordEyeCapture()
...
-            CoreInterface.RecordEyeData(EyeCaptures[index],hitType);
-
-
-            index = (index + 1) % CachedEyeCaptures;
+#if XRPF
+            if (XRPF.PrivacyFramework.Agreement.IsAgreementComplete && XRPF.PrivacyFramework.Agreement.IsSpatialDataAllowed)
+#endif
+            {
+                CoreInterface.RecordEyeData(EyeCaptures[index], hitType);
+                index = (index + 1) % CachedEyeCaptures;
```
Comment (AtifMahmud) [inline Runtime/Scripts/FixationRecorder.cs:989]: move out of the if

File: Runtime/Scripts/Cognitive3D_Manager.cs [hunk 2]
```diff
@@ -245,65 +244,89 @@ private void SetSessionProperties()
...
+                SetSessionProperty("xrpf.allowed.hardware.data", XRPF.PrivacyFramework.Agreement.IsHardwareDataAllowed);
+                SetSessionProperty("xrpf.allowed.bio.data", XRPF.PrivacyFramework.Agreement.IsBioDataAllowed);
+                SetSessionProperty("xrpf.allowed.spatial.data", XRPF.PrivacyFramework.Agreement.IsSpatialDataAllowed);
+                SetSessionProperty("xrpf.allowed.social.data", XRPF.PrivacyFramework.Agreement.IsSocialDataAllowed);
+            }
+#endif
+#endregion
+        }
+
+        public void SendHardwareDataAsSessionProperty()
```
Comment (AtifMahmud) [inline Runtime/Scripts/Cognitive3D_Manager.cs:269]: change it to private

File: Runtime/ExitPoll/ExitPoll.cs [hunk 3]
```diff
@@ -492,6 +492,17 @@ void IterateToNextQuestion()
                     return;
                 }
 
+                // Skip voice response if XRPF doesn't allow hardware data
+#if XRPF
```
Comment (AtifMahmud) [inline Runtime/ExitPoll/ExitPoll.cs:496]: get rid of this

**Review by calderarchinuk — APPROVED** [review]

**Pull request merged**